### DISCUSSION TITLE
Need to Improve the output after removing the watermark

### DISCUSSION THREAD

**Discussion by anieshvishnu** [ImageMagick/ImageMagick#3913] [Help] [answered]
Input

![11](https://user-images.githubusercontent.com/24503245/125037868-4fa5d280-e0b2-11eb-9e90-a9904088064f.png)

Output

![out](https://user-images.githubusercontent.com/24503245/125037941-68ae8380-e0b2-11eb-9abd-2d1e29a4b063.png)

I've removed the watermark by using `convert 1.png -blur x1 -threshold 50% out.png` Still the output has a darker shade where the watermark was present.

**Comment by snibgo** [comment, marked as answer]
The watermark has created black pixels. Some of these are isolated (they have white pixels to the north, south, east and west) and some have white pixels to the north and south but are adjacent to pixels that are black from the letters.

We can easily remove the isolated black pixels, using morphology. (Result x1.png in the command below.)

When a black pixel is adjacent to another black pixel, we cannot know whether it was made black by the watermark or because it was part of a letter. We can assume it should be removed, but this does damage some letters. (Result x2.png in the command below.)

Windows BAT script:
```
magick ^
  unofficialCopy.png ^
  ( +clone ^
    -morphology HMT 3x3:-,1,-,1,0,1,-,1,- ^
  ) ^
  -compose Lighten -composite ^
  +write x1.png ^
  ( +clone ^
    -morphology HMT 1x3:1,0,1 ^
  ) ^
  -compose Lighten -composite ^
  x2.png
```
More complex rules about removing black pixels could be devised.

**Comment by anieshvishnu** [comment]
@snibgo I've attached two images with watermark here we cant use morphology Hit and miss,  what should we use? Thank You.
![wm2](https://user-images.githubusercontent.com/24503245/137576005-09d76220-9413-4a93-9f61-34336b31685d.png)
![wm1](https://user-images.githubusercontent.com/24503245/137576006-24ad70d3-dcc1-40fa-b8e2-ac66a086bc1b.png)

**Comment by snibgo** [comment]
The first example seems to have a repeated pattern, a tiled image of black pixels spelling "Unofficial Copy" repeated at regular intervals. If that is correct, then you could crop out one of those tiled images then tile that across a white canvas. That would give you an image that is black where the watermark is, and white otherwise. Use that as a mask to turn watermarked pixels white.

This would turn white some of the pixels in the text "NORMAN BURNS P.O. Box". Correcting this would be very difficult.

The second image has no repeated pattern. What distinguishes the watermark from the original marks? The original marks are exactly black. The watermark is black and different shades of gray. You could use this to make a mask: a pixel is in the watermark if it is gray, or if it is black and adjacent to gray pixel.

That rule isn't quite sufficient, because some black pixels are entirely surrounded by other black pixels, and this can occur in both the watermark and the original marks.
